### PR TITLE
fix: Reference name even if not unique, provided it has only one 01 l…

### PR DESCRIPTION
**Pull request by ap891843**
…evel

If there is only one 01 level with a given name, that name can be referenced even if it is not unique when the QUALIFY(EXTEND) option is in effect.
Ref -
https://www.ibm.com/docs/en/cobol-zos/6.3?topic=reference-qualification

Signed-off-by: Aman Prashant <aman.prashant@broadcom.com>

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/CompilerDirectiveNode.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.eclipse.lsp.cobol.common.model.tree.Node;
+import org.eclipse.lsp.cobol.core.CobolParser;
+
+/** Node describing compiler directive in a cobol document. */
+@Slf4j
+public class CompilerDirectiveNode extends Node {
+  @Getter private final CobolParser.CompilerOptionsContext directiveContext;
+
+  public CompilerDirectiveNode(
+      Locality location, CobolParser.CompilerOptionsContext directiveContext) {
```
Comment (Nurkambay) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/CompilerDirectiveNode.java:30]: Please get rid of `CobolParser` in the Node. AST must not depend on the parser, in the future, we can use different parsers. Also, all nodes will be moved to the common module that has no dependency on ANTLR4

Comment (ap891843) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/CompilerDirectiveNode.java]: Yes, make sense :)
Updated

File: server/common/src/main/java/org/eclipse/lsp/cobol/common/processor/CompilerDirectiveName.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  ADATA {
+    @Override
+    public Optional<CompilerDirectiveOption> getDirectiveOption(String directiveText) {
+      return Optional.empty();
+    }
+  },
+  QUALIFY {
+    @Override
+    public Optional<CompilerDirectiveOption> getDirectiveOption(String directiveText) {
+      if (directiveText.contains("QUALIFY(EXTEND)"))
```
Comment (Nurkambay) [inline server/common/src/main/java/org/eclipse/lsp/cobol/common/processor/CompilerDirectiveName.java:34]: Maybe something like:
```java
return Optional.of(new CompilerDirectiveOption<>(true, this).filter(cd -> cd.getDirectiveText(contains("QUALIFY(EXTEND)"));
```
?

Comment (ap891843) [reply server/common/src/main/java/org/eclipse/lsp/cobol/common/processor/CompilerDirectiveName.java]: Nice :). Updated

**Pull request merged**